src/path.c

```diff
@@ -56,7 +56,7 @@ alloc_array(Py_ssize_t count)
         PyErr_NoMemory();
         return NULL;
     }
-    if (count > (SIZE_MAX / (2 * sizeof(double))) - 1 ) {
+    if ((unsigned long)count > (SIZE_MAX / (2 * sizeof(double))) - 1 ) {
```

Comment:
Immediately before this, the code `return`s if `count < 0`, so it is safe to cast it to unsigned here.